core/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -326,4 +322,8 @@ public Version indexVersionCreated() {
     public QueryParseContext parseContext() {
         return this.parseContext;
     }
+
+    public Client client() {
```

Comment:
maybe we can avoid exposing the Client and instead expose only one new method with the needed functionality in INdexQueryParseService and simply proxy it in the context? That would prevent us from spreading around the big client dependency